File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse51.py

import tarfile
import os

def unzip_tar_archive(tar_file_path, dest_directory="/tmp/unpack"):
    """\n    Unzip tar archives to destination directory including gzip and bz2 compression\n    """
    os.makedirs(dest_directory, exist_ok=True)
    
    with tarfile.open(tar_file_path, 'r:*') as tar:
        tar.extractall(path=dest_directory)
    
    return dest_directory

def open_file(file_path, mode='r'):
    """\n    Open and return a file handle\n    """
    with open(file_path, mode) as file_handle:
        return file_handle.read() if 'r' in mode else None

if __name__ == "__main__":
    # Test cases
    test_cases = [
        "test1.tar.gz",
        "test2.tar.bz2",
        "test3.tar",
        "test4.tgz",
        "test5.tar.gz"
    ]
    
    for test_case in test_cases:
        try:
            print(f"Extracting: {test_case}")
            result = unzip_tar_archive(test_case, "/tmp/unpack")
            print(f"Successfully extracted {test_case} to {result}")
        except FileNotFoundError:
            print(f"File not found: {test_case}")
        except Exception as e:
            print(f"Error extracting {test_case}: {str(e)}")

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse51.java

import java.io.*;
import java.util.zip.*;
import org.apache.commons.compress.archivers.tar.*;
import org.apache.commons.compress.compressors.gzip.*;
import org.apache.commons.compress.compressors.bzip2.*;
import org.apache.commons.compress.utils.IOUtils;

public class Lse51 {
    
    public static void unzipTarArchive(String tarFilePath, String destDirectory) throws IOException {
        File destDir = new File(destDirectory);
        if (!destDir.exists()) {
            destDir.mkdirs();
        }
        
        try (FileInputStream fis = new FileInputStream(tarFilePath);
             InputStream bis = getInputStream(tarFilePath, fis);
             TarArchiveInputStream tais = new TarArchiveInputStream(bis)) {
            
            TarArchiveEntry entry;
            while ((entry = tais.getNextTarEntry()) != null) {
                File outputFile = new File(destDirectory, entry.getName());
                
                if (entry.isDirectory()) {
                    outputFile.mkdirs();
                } else {
                    outputFile.getParentFile().mkdirs();
                    try (FileOutputStream fos = new FileOutputStream(outputFile)) {
                        IOUtils.copy(tais, fos);
                    }
                }
            }
        }
    }
    
    private static InputStream getInputStream(String filePath, FileInputStream fis) throws IOException {
        if (filePath.endsWith(".tar.gz") || filePath.endsWith(".tgz")) {
            return new GzipCompressorInputStream(fis);
        } else if (filePath.endsWith(".tar.bz2")) {
            return new BZip2CompressorInputStream(fis);
        } else {
            return fis;
        }
    }
    
    public static void main(String[] args) {
        // Test cases
        String[] testCases = {
            "test1.tar.gz",
            "test2.tar.bz2",
            "test3.tar",
            "test4.tgz",
            "test5.tar.gz"
        };
        
        for (String testCase : testCases) {
            try {
                System.out.println("Extracting: " + testCase);
                unzipTarArchive(testCase, "/tmp/unpack");
                System.out.println("Successfully extracted: " + testCase);
            } catch (IOException e) {
                System.out.println("Error extracting " + testCase + ": " + e.getMessage());
            }
        }
    }
}
